core/Azure.Mcp.Core/src/Areas/Server/Commands/ServiceStartCommand.cs

```diff
@@ -269,20 +298,17 @@ private static void ValidateNamespaceAndToolMutualExclusion(string[]? namespaces
     /// <summary>
     /// Validates that the outgoing authentication strategy is compatible with the hosting mode.
     /// </summary>
-    /// <param name="runAsRemoteHttpService">Whether the server is running as a remote HTTP service.</param>
-    /// <param name="outgoingAuthStrategy">The outgoing authentication strategy.</param>
     /// <param name="commandResult">Command result to update on failure.</param>
-    private static void ValidateOutgoingAuthStrategy(
-        bool runAsRemoteHttpService,
-        OutgoingAuthStrategy outgoingAuthStrategy,
-        CommandResult commandResult)
+    private static void ValidateOutgoingAuthStrategy(CommandResult commandResult)
     {
-        // UseOnBehalfOf is only valid when running as a remote HTTP service
-        if (!runAsRemoteHttpService && outgoingAuthStrategy == OutgoingAuthStrategy.UseOnBehalfOf)
+        var outgoingAuthStrategy = commandResult.GetValueOrDefault<OutgoingAuthStrategy>(ServiceOptionDefinitions.OutgoingAuthStrategy.Name);
+        if (outgoingAuthStrategy == OutgoingAuthStrategy.UseOnBehalfOf)
```

Comment:
Why is `!runAsRemoteHttpService` check removed? Do we support OBO when running in `stdio` mode?

Reply:
OBO is not supported in `stdio` because the MCP spec does not have a mechanism for an MCP client to authenticate to the MCP server using that transport. In that case, the "sign in" is happening within the MCP server which is the `UseHostingEnvironmentIdentity` option.